wear_os_bible_ereader/lib/app/view/app.dart

```diff
@@ -142,34 +142,153 @@ class BookDetailsPage extends StatelessWidget {
         return context.read<PositionCubit>().popMenu();
       },
       child: Scaffold(
-        appBar: AppBar(title: const Text('Details')),
-        body: Padding(
-          padding: const EdgeInsets.all(8),
-          child: BlocBuilder<PositionCubit, PositionState>(
-            builder: (context, state) {
-              return state.loadingDocument
-                  ? Text('Loading ${state.bookTitle}')
-                  : Stack(
-                      children: [
-                        // Always build EpubView (and sometimes hide it behind menu)
-                        // since it seems to handle all the controller loading and
-                        // listenable notification logic. I originally tried having
-                        // table of contents in a separate page (instead of a Stack)
-                        // but it would never get notified that listenable values
-                        // changed or that the document finished loading. I'm not
-                        // sure if this is the best approach, but it seems to work
-                        // well having an EpubView always in the widget tree (even
-                        // if it's not visible).
-                        EpubView(controller: epubController),
-                        if (state.chapterIndex == null) ...[
-                          Container(
-                              color: Theme.of(context).colorScheme.background),
-                          TableOfContents(epubController: epubController),
-                        ],
-                      ],
-                    );
-            },
-          ),
+        body: CustomScrollView(
+          slivers: [
+            SliverAppBar(
+              title: Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  IconButton(
+                    icon: const Icon(Icons.home),
+                    onPressed: () {},
+                  )
+                ],
+              ),
+              floating: true,
+              automaticallyImplyLeading: false, // No back arrow.
+            ),
+
+            /// UNCOMMENTING THIS AND COMMENTING OUT OTHER SliverToBoxAdapter
+            /// SHOWS HOW THE floating AppBar can work, but it won't work for
+            /// EpubView's nested ScrollablePositionedList.
```

Comment:
CustomScrollView and SliverAppBar make the AppBar hide on scroll like I wanted, but EpubView's nested ScrollablePositionedList can't work (see comment on line 198).

ScrollablePositionedList is required so we can scroll to a specific chapter etc.

Backup plan is to make "app bar" (home and settings icons) show/hide with a double tap like Gospel Library. I may be able to use a gestureDetector to handle single-tap scrolls and double-tap show/hide menu but ignore swipes (let list scroll on swipe).

Reply:
https://stackoverflow.com/questions/53941362/flutter-how-can-i-dynamically-show-or-hide-app-bars-on-pages may be helpful

Reply:
Double-tap implemented in https://github.com/KyleFin/wear-bible-ereader/pull/13